### DISCUSSION TITLE
Count members with role

### DISCUSSION THREAD

**Discussion by DevAreaServer** [Discord4J/Discord4J#948] [Q&A] [answered]
Hey !

I'm looking for a while a method on Role "getMemberCount", for know the count of member with this role.

I don't know if I didn't saw the method or there is no method for this :/

Thanks for help me !

![image](https://user-images.githubusercontent.com/86724981/124014078-5a41d700-d9e3-11eb-9fcf-e90bfa84f0b8.png)
Like this

**Comment by NovaFox161** [comment, marked as answer]
I would recommend joining the support server and asking in the help channel dedicated to using the library as its much quicker to get a response.

However, to answer your question, you would have to to through all the members of the guild and check how many people have the role.

**Reply by DevAreaServer** [reply]
Yes that's what I have done until now but I was wondering if there was more efficient than recovering the whole list of members :/

Thank's for the answer !

I'll check on discord !

**Comment by darichey** [comment]
As @NovaFox161 said, you'll have to do this manually. Discord does not provide a way to do it. We could potentially consider a utility method for it, but it seems a little niche. Anyways, here's an example of how it might look.
```java
Mono<Long> countMembersWithRole(Guild guild, Snowflake roleId) {
    return guild.getMembers()
            .filter(member -> member.getRoleIds().contains(roleId))
            .count();
}
```

It's worth noting that if you want to do something like you showed in that picture, it would be inefficient to just loop over all of the roles and call the above method (especially if the members need to be requested and/or the guild is very large). Instead, you can get all of the counts at once, like so:
```java
Mono<Map<Snowflake, Long>> getRoleCounts(Guild guild) {
    return guild.getMembers()
            .flatMap(member -> Flux.fromIterable(member.getRoleIds()))
            .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()))
}
```

Also note that for either of the above solutions, you will need the `GUILD_MEMBERS` [intent](https://discord.com/developers/docs/topics/gateway#gateway-intents).
